### DISCUSSION TITLE
Source state of polarisation - Polarisation ray tracing

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2952] [Q&A]
Hello there friends from the MEEP community,

I have used MEEP a couple of times in order to model the radiation from point emitters embedded within polymer layers. In general, this works for me, I am able to "translate" the generated far-field components into a set of rays which I then import into a ray-tracing software to perform ray-tracing calculations.

The workflow that I follow in this case is the following:
1. First I extract far-field components via a near to far-field transformation.
2. From these, I compute the Poynting vector. Here the assumption that I always take is that the emitted light can be described as being originated from a single point and this is something that I have verified by comparing the direction of the Poynting vector's with the directions normal to a semi-sphere surface. 
3. Using the Poynting vectors, I can then define direction cosines which represent the "ray" directions and the Poynting vector's projection along these "ray" directions represent my "ray" intensities. 
4. This is what I then import into the ray-tracer (direction cosines and local ray intensities)

The thing is that now I would like to perform polarisation-ray tracing analysis and for this, I need to first define the polarisation state of my field. 
From what I have read and what I know, the local state of polarisation is expressed in terms of two orthogonal field components that are defined in the plane perpendicular to the local k vector (or Poynting vector direction). However, from MEEP, the field components that I obtain ($E_x$, $E_y$ and $E_z$) are defined with respect to the global coordinate system.

From this, my first idea would be to re-express the field vector $\vec E =[E_x,E_y,E_z]$ in terms of locally defined axes $x_{local}$ and $y_{local}$ which span the plane perpendicular to k at each point, to obtain two local field components $E_{x_{local}}$ and $E_{y_{local}}$ and then use these field components to estimate the polarisation Jones vectors or stokes parameters. 

However, it is this last part that I am not totally sure if it would be consistent. For example, would it matter that the local axes directions vary across the field evaluation points? How would this affect the state of polarisation estimation?

Any comments or thoughts about this?

**Comment by stevengj** [comment]
In a ray tracing context, where you are far enough away from the source to locally treat the field as a planewave, then you just want to compute the two E components perpendicular to the Poynting vector, which is a simple geometric transformation from Cartesian coordinates.

There's still the question of what basis you use to define the polarization state, but presumably that's specified by your ray-tracing software?
